modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/_diffs/plugins/toolbarbuttons/plugin.js

```diff
@@ -13,14 +13,246 @@
  */
 
 (function () {
+	const HEADING_BOTH = 'Both';
+	const HEADING_COL = 'Column';
+	const HEADING_NONE = 'None';
+	const HEADING_ROW = 'Row';
+
 	const pluginName = 'toolbarbuttons';
 
 	if (CKEDITOR.plugins.get(pluginName)) {
 		return;
 	}
 
+	/**
+	 * Creates a new CKEDITOR.dom.element using the passed tag name.
+	 *
+	 * @instance
+	 * @memberof CKEDITOR.Table
+	 * @protected
+	 * @method _createElement
+	 * @param {String} name The tag name from which an element should be created
+	 * @param {Object} editor The CKEditor instance.
+	 * @return {CKEDITOR.dom.element} Instance of CKEDITOR DOM element class
+	 */
+	function _createElement(name, editor) {
+		return new CKEDITOR.dom.element(name, editor.document);
+	}
+
+	/**
+	 * Returns which heading style is set for the given table.
+	 *
+	 * @instance
+	 * @memberof CKEDITOR.Table
+	 * @method getHeading
+	 * @param {CKEDITOR.dom.element} table The table to gather the heading from. If null, it will be retrieved from the current selection.
+	 * @param {Object} editor The CKEditor instance.
+	 * @return {String} The heading of the table. Expected values are `CKEDITOR.Table.NONE`, `CKEDITOR.Table.ROW`, `CKEDITOR.Table.COL` and `CKEDITOR.Table.BOTH`.
+	 */
+	function getHeading(table, editor) {
+		table = table || getFromSelection(editor);
+
+		if (!table) {
+			return null;
+		}
+
+		const rowHeadingSettings = table.$.tHead !== null;
+
+		let colHeadingSettings = true;
+
+		// Check if all of the first cells in every row are TH
+
+		for (let row = 0; row < table.$.rows.length; row++) {
+
+			// If just one cell isn't a TH then it isn't a header column
+
+			const cell = table.$.rows[row].cells[0];
+
+			if (cell && cell.nodeName.toLowerCase() !== 'th') {
+				colHeadingSettings = false;
+				break;
+			}
+		}
+
+		let headingSettings = HEADING_NONE;
+
+		if (rowHeadingSettings) {
+			headingSettings = HEADING_ROW;
+		}
+
+		if (colHeadingSettings) {
+			headingSettings =
+				headingSettings === HEADING_ROW ? HEADING_BOTH : HEADING_COL;
+		}
+
+		return headingSettings;
+	}
+
+	/**
+	 * Retrieves a table from the current selection.
+	 *
+	 * @instance
+	 * @memberof CKEDITOR.Table
+	 * @method getFromSelection
+	 * @param {Object} editor The CKEditor instance.
+	 * @return {CKEDITOR.dom.element} The retrieved table or null if not found.
+	 */
+	function getFromSelection(editor) {
+		let table;
+		const selection = editor.getSelection();
+		const selected = selection.getSelectedElement();
```

Comment:
Can we name this variable `selectedElement`, to be consistent with other plugins? `selection` and `selected` are way too similar.

Reply:
yes, it makes totally sense :)